96.md

```diff
@@ -0,0 +1,325 @@
+NIP-96
+======
+
+HTTP File Storage Integration
+-----------------------------
+
+`draft` `optional` `author:arthurfranca` `author:Semisol` `author:staab`
+
+## Motivation
+
+This NIP defines a simple unified REST API for HTTP file storage servers so to integrate them to the nostr network.
+The API will enable nostr users to upload files and later reference them by url on nostr notes.
+
+The spec DOES NOT use regular nostr events through websockets for
+storing, requesting nor retrieving data because, for simplicity, the server
+will not have to learn anything about nostr relays.
+
+## Server Adaptation
+
+File storage servers wishing to be accessible by nostr users should opt-in by making available
+an https route at `/nip96(/<sha256-file-hash>)` for handling files as described below.
+
+### List of Supporting File Storage Servers
+
+| Name | Domain | Upload Cost | NIP-60 Fee | File Retention Policy |
+| ---- | ------ | ----------- | ---------- | --------------------- |
+| -    | -      | -           | -          | ❌️                    |
+
+## Upload
+
+A file can be uploaded one at a time to `https://server.domain/nip96` as `multipart/form-data` content type using `POST` method with the file object set to the `file` form data field.
```

Comment:
```suggestion
A file can be uploaded one at a time to `https://server.domain/.well-known/nip96` as `multipart/form-data` content type using `POST` method with the file object set to the `file` form data field.
```

It's a big assumption that servers can simply add a new route. For example, Twitter would have to delete this user's account: https://twitter.com/nip96

There are all kinds of problems with exposing paths on the top-level. People will have to update their Service Workers and caches. At least `.well-known` is well-known, and usually already has some special handling.

Curious what @melvincarvalho thinks about this.

Reply:
Also, I think this NIP can evolve by having relays add nip96 to their supported_nips and expose the nip96 endpoint. Then clients can select a relay with nip96 support to upload files to.

Reply:
isn't .well-known more used for metadata stuff? not really where you'd put an upload route.

what about putting a nip96 object in /.well-known/nostr.json that contains info about the routes?

Reply:
Using  /.well-known/nostr.json as config now